doc/release-notes/9340-payara5to6.md

```diff
@@ -0,0 +1,95 @@
+## Upgrade from Payara 5 to Payara 6
+
+1. Download Payara 6.2023.7 as of this writing:
+
+   `curl -L -O https://nexus.payara.fish/repository/payara-community/fish/payara/distributions/payara/6.2023.7/payara-6.2023.7.zip`
+
+1. Unzip it to /usr/local (or your preferred location)
+
+   `sudo unzip payara-6.2023.7.zip -d /usr/local/`
+
+1. Change ownership of the unzipped Payara to your "service" user ("dataverse" by default)
+
+   `sudo chown -R dataverse /usr/local/payara6`
+
+1. Undeploy Dataverse (if deployed; version 5.14 is assumed in the example below)
+
+   `sudo /usr/local/payara5/bin/asadmin list-applications`
+
+   `sudo /usr/local/payara5/bin/asadmin undeploy dataverse-5.14`
+
+1. Stop Payara 5
+
+   `sudo /usr/local/payara5/bin/asadmin stop-domain`
+
+1. Copy Dataverse-related lines from Payara 5 to Payara 6 domain.xml
+
+   `sudo cp /usr/local/payara6/glassfish/domains/domain1/config/domain.xml /usr/local/payara6/glassfish/domains/domain1/config/domain.xml.orig`
+
+   `sudo egrep 'dataverse|doi' /usr/local/payara5/glassfish/domains/domain1/config/domain.xml > lines.txt`
+
+   `sudo vi /usr/local/payara6/glassfish/domains/domain1/config/domain.xml`
+
+   The lines will be part of three sections, shown below, but your content will vary.
+   
+   Section 1: mail resource (under `<resources>`)
+
+   ```
+   <mail-resource auth="false" host="localhost" from="do-not-reply@dev1.dataverse.org" user="dataversenotify" jndi-name="mail/notifyMailSession"></mail-resource>
+   ```
+
+   Section 2: system properties (under `<server name="server" config-ref="server-config">`)
+    
+   ```
+   <system-property name="dataverse.db.user" value="dvnuser"></system-property>
+   <system-property name="dataverse.db.host" value="localhost"></system-property>
+   <system-property name="dataverse.db.port" value="5432"></system-property>
+   <system-property name="dataverse.db.name" value="dvndb"></system-property>
+   <system-property name="dataverse.db.password" value="dvnsecret"></system-property>
```

Comment:
as-setup.sh creates these as aliases: https://github.com/IQSS/dataverse/blob/develop/scripts/installer/as-setup.sh#L75
so existing installations may not have `dataverse.db.password` but will instead want to create a new password alias.

Reply:
Good point. We should tell people to create a password alias. More secure.